### ISSUE TITLE
CORS not accepting requests from set Origins when set for all routes

### ISSUE THREAD

**Issue by damusix** [hapijs/hapi#3789]
#### Are you sure this is an issue with the hapi core module or are you just looking for some help?
I might just need some help, but I haven't been able to find a proper explanation for globally configuring CORS. I have searched in closed issues, stack exchange, and scoured the documentation. I haven't found a clear explanation.

#### What are you trying to achieve or the steps to reproduce?
- `mkdir cors; cd cors; npm i hapi http-server;`
- Create an `index.html` page that includes Axios and send a request to `localhost:4055/auth`
- Create an `index.js` file with the following code
- Run `node ./index.js`
- Open `index.html` using `http-server` or something easy (different URL to trigger CORS)
    - `./node_modules/.bin/http-server -p 3000`
    - http://localhost:3000

The issue is exactly the same even if I use `origin: ['localhost', 'localhost:3000']` when developing local

**index.js**
```js
const Hapi = require('hapi');

// I reverse proxy domain to localhost:3000 with nginx
// server = new Hapi.Server({ port: 4055, routes: { cors: { origin: ['http://react.pinguino'] } } });
server = new Hapi.Server({ port: 4055, routes: { cors: { origin: ['localhost', 'localhost:3000'] } } });


(async function() {

    server.ext('onRequest', (request, h) => {

        console.log('INFO:', request.info);
        console.log('HEADERS:', request.headers);

        return h.continue
    })

    await server.start()


})()
```

**index.html**
``` html
<html>
<head>
<script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.18.0/axios.min.js"></script>

<script> axios.put('http://locahost:4055/auth') </script>
</head>
<body></body>
```

#### What was the result you received?
On the frontend: `Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://react.pinguino' is therefore not allowed access. The response had HTTP status code 404`
<img width="582" alt="screen shot 2018-05-04 at 3 49 41 pm" src="https://user-images.githubusercontent.com/4218587/39650046-b222d6c0-4fb4-11e8-9220-b51541267554.png">

On the server:
<img width="840" alt="screen shot 2018-05-04 at 3 45 40 pm" src="https://user-images.githubusercontent.com/4218587/39650035-acd8edda-4fb4-11e8-8a57-1c1d29132da2.png">

#### What did you expect?
A JSON reponse from my server

#### Context

* *node version*: v8.11.1
* *hapi version*: 17.4.0
* *os*: OSX High Sierra
* *any other relevant information*:

On my real application, I have attempted this with the following configuration to rule out rejected headers:
``` js
Hapi.Server({
        routes: {

            security: true,
            cors: {
                origin: [
                    'http://localhost:3000',
                    'http://localhost',
                    'http://react.pinguino'
                ],
                additionalHeaders: [
                    'Access-Control-Allow-Origin',
                    'Access-Control-Request-Method',
                    'Allow-Origin',
                    'Origin',
                    'access-control-allow-origin',
                    'access-control-request-method',
                    'allow-origin',
                    'origin',
                ]
            }
        }
})
```

**Comment by Phonbopit** [comment]
@damusix  Your route `PUT /auth` does exist?
I'm facing the same problem, but I resolve my problem because I don't have a route that matches with frontend request e.g. I request `POST /auth/login` but the route doesn't exist in server, then It's return 404.

I see a code in `lib/cors.js` here https://github.com/hapijs/hapi/blob/master/lib/cors.js#L100-L102 Not sure it helps or not

**Comment by damusix** [comment]
@Phonbopit This is true, however, when I do a `POST` it gives a proper `404` without the CORS error, but `PUT` gives me a CORS error. I am experiencing this even when the route exists. On my app, which DOES have `PUT` to `/auth`, I get the exact same issue.

**Comment by hueniverse** [comment]
The issue with 404 handling is being addressed by #3792. Otherwise, please post requests for help in the discuss repo.
